### ISSUE TITLE
Version 2.3.1 does not recognize pluralized enums

### ISSUE THREAD

**Issue by foncho** [lwe/simple_enum#137]
Hi there, I have an application, which have a model called Tax. There are two tax types: impositives and retentives. Here is my model declaration:

```
class Tax < ActiveRecord::Base
  as_enum :tax_type, %i{ retentive impositive }
end
```

So, when I want to get all impositives or retentices

`Tax.impositives`

I've got the next:

`NoMethodError: undefined method `impositives' for #<Class:0x007ffad03e71f0>`

Any idea?

**Comment by lwe** [comment]
Hi @foncho, this is weird, does it work without pluralizing - or by explicitly setting pluralize_scopes: true?

**Comment by lwe** [comment]
@drobny, do you have any idea if this could be a regression from adding the pluralize_scopes option?

**Comment by drobny** [comment]
@foncho @lwe potentially, I will do my best to have a look at this today or after work

**Comment by foncho** [comment]
@lwe yes, it’s so weird. Singularize option works fine, but not the pluralize one.

**Comment by foncho** [comment]
Thanks @drobny

**Comment by drobny** [comment]
@foncho Just had a quick look and not managed to replicate it just by spinning up a simple app and mirroring your example 😕 

```
class Tax < ApplicationRecord
  as_enum :tax_type, %i{ retentive impositive }
end
```

```
2.2.8 :001 > Tax.impositives
  Tax Load (0.1ms)  SELECT  "taxes".* FROM "taxes" WHERE "taxes"."tax_type_cd" = ? LIMIT ?  [["tax_type_cd", 1], ["LIMIT", 11]]
 => #<ActiveRecord::Relation []>
```

Is it possible for you to link to the app you are using or other simple_enum config you have put in an initializer or something?

**Comment by lwe** [comment]
@drobny, thanks for checking and providing the support - much appreciated! Checked the code again and I also agree with the assessment, that this probably has another impact or root cause. Can you provide more details, as well:

- Rails version
- Used database (mySQL, postgres?)
- Other gems being used?

**Comment by foncho** [comment]
@drobny nothing but the standard configuration. No initializer or special config added.
@lwe here you have the data:

- Rails v4.2.10
- PostgresSQL database engine
- As a big management application, I have used several gems. Here you have my Gemfile:
```
gem 'rails', '~> 4'
gem 'puma'

gem 'pg', '< 1.0.0'
gem 'dirty_associations'
gem 'activerecord-session_store'

gem 'haml-rails'

gem 'jquery-rails'
gem 'sass-rails'
gem 'less-rails'
gem 'uglifier'
gem 'libv8'
gem 'therubyracer', platform: :ruby
gem 'sprockets', '3.6.3'
gem 'sprockets-rails', '~> 2.3'

gem 'momentjs-rails'

gem 'ckeditor'

gem 'devise'
gem 'simple_roles'
gem 'cancancan'
gem 'paper_trail'

gem 'simple_form'
gem 'nested_form'
gem 'kaminari'
gem 'kaminari-i18n'
gem 'kaminari-bootstrap', '~> 3'

gem 'ransack', '1.7.0'

gem 'country_select', github: 'stefanpenner/country_select'
gem 'localized_country_select'
gem 'numbers_and_words'

gem 'jquery-fileupload-rails'

gem 'paperclip'
gem 'remotipart'
gem 'mini_magick'

gem 'wicked_pdf'
gem 'wkhtmltopdf-binary'

gem 'color'

gem 'whenever', require: false

gem 'simple_enum'
gem 'request_store'

gem 'acts_as_xlsx', git: "git://github.com/straydogstudio/acts_as_xlsx.git"
gem 'axlsx_rails'

gem 'humanize', path: 'lib/gems/humanize'

gem 'spring', group: :development

group :development, :test do
  gem 'byebug'
  gem 'ruby-debug-ide'
  gem 'debase'
  gem 'pry-rails'
  gem 'faker'
  gem 'rspec-rails'
  gem 'shoulda-matchers'
  gem 'database_cleaner'
  gem 'factory_bot_rails'
  gem 'capybara-rails'
end

gem 'passenger'
gem 'dalli'
```

**Comment by drobny** [comment]
Thanks both will do my best to have a look at this again tomorrow

**Comment by lwe** [comment]
@foncho, any chance you can point as to a minimal git repo where we can reproduce the issue? (Ideally with sqlite)?

**Comment by drobny** [comment]
@foncho @lwe I made a repo https://github.com/drobny/simple-enum-test that matches the Gemfile as closely as possible. The only differences are using sqlite3 and commenting out the local `humanize` gem. Unfortunately I still can't replicate the issue you are seeing:

```
Loading development environment (Rails 4.2.10)
[1] pry(main)> Tax.impositives
  Tax Load (0.2ms)  SELECT "taxes".* FROM "taxes" WHERE "taxes"."tax_type_cd" = ?  [["tax_type_cd", 1]]
=> []
```

Feel free to clone and have a play around with my repo, I might have missed something obvious. Is there anything else declared in your `Tax` model? What version of ruby are you using?

**Comment by gordysc** [comment]
I've managed to hit this exact same issue in our application as well in the development environment.  I've thrown together a repo that reproduces the issue here:
https://github.com/gordysc/simple_enum_example

I'm using Rails 5.1.5 on my Mac (10.13.3)

**Comment by drobny** [comment]
@gordysc thanks for taking the time to setup the repo, I will do my best to have a look into it this weekend and see if I can figure out how to fix it

**Comment by lwe** [comment]
👍 thank y'all for the help and the reproduction repo!

**Comment by drobny** [comment]
@foncho @gordysc thanks to your help I managed to replicate the issue and have https://github.com/lwe/simple_enum/pull/138 which should fix the issue you are currently seeing. Would you mind bundling from my branch and confirm if this does fix it for you?

**Comment by foncho** [comment]
@lwe @drobny I've tried again with the upgrade to 2.3.1 and now it works, so I think you can successfully close the issue. Thanks a lot guys!

**Comment by gordysc** [comment]
Agreed, thanks for getting this in!

**Comment by espen** [comment]
I faced this issue with 2.3.1 but it looks like it is fixed with 2.3.2 so perhaps this issue should be closed?

**Comment by lwe** [comment]
Thanks @espen, will close it.
